[Streams article] Various fixes (#4713)
GoogleChrome/web.dev: src/site/content/en/blog/streams/index.md

authors:
  - thomassteiner
date: 2021-02-19

> updated: 2021-02-22

Reviewer: ```suggestion
updated: 2021-02-25
```

### Readable stream code samples

The code sample below shows all the steps in action. You first create a `ReadableStream` that in its

> `underlyingSource` argument (that is, the `TimestampSource` class) defines a `start` method

Reviewer 2: ```suggestion
`underlyingSource` argument (that is, the `TimestampSource` class) defines a `start()` method
```

Reviewer 2: I think this tries to convey too much in a single sentence, which I'm having trouble following. There doesn't have to be a 1 to 1 relationship between lines of text and lines of code, but it helps if single gestures in the code get at least their own suggestions.

that tells the stream's `controller` to
`enqueue()` a timestamp every second during ten seconds and then to `close()` the stream. You consume this
stream by creating a reader via the `getReader()` method and calling `read()` until the stream is
`done`.

```js
class TimestampSource {

>   #interval

Reviewer 2: Please stick to plain JavaScript.

Reviewer: This is the new [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) notation.

### The `queuingStrategy`

The second, likewise optional, argument of the `ReadableStream()` constructor is `queuingStrategy`.
It is an object that optionally defines a queuing strategy for the stream, which takes one
parameter:

- `highWaterMark`: A non-negative number of bytes indicating the high water mark of the stream using this queuing strategy.
  This is used to determine backpressure, manifesting via the appropriate `ReadableByteStreamController.desiredSize` property.
  It also governs when the underlying source's `pull()` method is called.

> Note that a queuing strategy for a readable byte stream does not have a `size(chunk)` function.
> The size of each chunk is always determined by its `byteLength` property.

Reviewer: ```suggestion
{% Aside %}
  A queuing strategy for a readable byte stream does not have a `size(chunk)` function.
  The size of each chunk is always determined by its `byteLength` property.
{% endAside %}  
```

We have a special "note" aside widget.

### The `queuingStrategy`

The second, likewise optional, argument of the `ReadableStream()` constructor is `queuingStrategy`.
It is an object that optionally defines a queuing strategy for the stream, which takes one
parameter:

- `highWaterMark`: A non-negative number of bytes indicating the high water mark of the stream using this queuing strategy.
  This is used to determine backpressure, manifesting via the appropriate `ReadableByteStreamController.desiredSize` property.
  It also governs when the underlying source's `pull()` method is called.

{% Aside %}

>   A queuing strategy for a readable byte stream does not have a `size(chunk)` function.

Reviewer 2: I want to make sure there isn't something else to explain. Why would I expect a `size(chunk)` function?

Reviewer: Because all the other stream variants have one to set a custom queuing strategy.

Reviewer 2: maybe add , "like other stream types do"? 

This is the conceptual version of a broken link. If I'm not already familiar with streams, I can't work out for myself why this information is here or why I care.

Author: How about:
> Unlike queuing strategies for other stream types, a queuing strategy for a readable byte stream does not have a `size(chunk)` function.

Reviewer: I went with this variant in https://github.com/GoogleChrome/web.dev/pull/4713/commits/cdab862037056c1084b3a841c7cb55a421ffe1fa.
